### PR TITLE
update post results

### LINKED ISSUE #293: update the votes feature
## Enhancement 
 we want to develop a better approach when working with the votes.

### what to do
 we will align with the backend team so they will provide us with specific API data, so we can do the needed logic in the front end and add the votes feature properly

### PR DESCRIPTION
**Pull request by atwfk**
closes #293
- removed `transformAuthorizedPosts` as it's useless now and the backend is not sending `vote_count` property on each post
- added `voted` property for IOptions interface 
- added the needed adjustments for the post view comps tests and storybook
- removed  `optionCheckedId` state from the post view componsnts types
- renamed `https://pickify-posts-be-dev.pickify.net/api` api to `https://pickify-posts-be-dev.m3ntorship.net/api`
- fixed sending default userId if app renders in the server and added cookie for user id

this pr contains 33 changed files as i had to add the `voted` property for all the nested post view comps,
and removed `optionCheckedId` from all the nested types
and you will notice that most of the changed files contains only 1 line change

### CODE REVIEW

File: modules/HomePage/components/Posts/PostsHelpers.ts [hunk 1]
```diff
@@ -30,38 +30,41 @@ export const transformPostsMedia = (
...
-          return { id, body, media };
-        });
-        return { ...group, options: newOptions };
-      });
-      return { ...post, options_groups: { groups } };
-    },
-  );
-  return authorizedPosts;
-};
+// export const transformAuthorizedPosts = (
```
Comment (Nilegfx) [inline modules/HomePage/components/Posts/PostsHelpers.ts:33]: remove

File: modules/shared/api/getPosts.api.ts [hunk 2]
```diff
@@ -4,24 +4,52 @@ import type { IGetPosts } from './IGetPosts';
...
-export const getPosts = async (): Promise<IGetPosts.IData> => {
-  const notFound = 404;
+export const getPosts = async (userId: string): Promise<IGetPosts.IData> => {
   return postsApi
-    .getPosts()
+    .getPosts(undefined, undefined, {
+      headers: { Authorization: `Bearer ${userId}` },
+    })
     .then(({ data }) => {
+      // console.dir(data, { depth: null });
```
Comment (Nilegfx) [inline modules/shared/api/getPosts.api.ts:13]: remove

File: modules/shared/api/getPosts.api.ts [hunk 3]
```diff
@@ -4,24 +4,52 @@ import type { IGetPosts } from './IGetPosts';
...
       return { data };
     })
     .catch((error: AxiosError) => {
+      const { response } = error as { response?: { status: number } };
+      const notFound = 404;
```
Comment (Nilegfx) [inline modules/shared/api/getPosts.api.ts:18]: 500 not 404

File: modules/shared/api/getPosts.api.ts [hunk 4]
```diff
@@ -4,24 +4,52 @@ import type { IGetPosts } from './IGetPosts';
...
           error: true,
-          errorCode: error.response ? error.response.status : notFound,
+          errorCode: response ? response.status : notFound,
           message: error.message,
         },
       };
     });
 };
 
+// export const getPostsTemp = async (
```
Comment (Nilegfx) [inline modules/shared/api/getPosts.api.ts:29]: remove

File: modules/shared/api/postsApi.api.ts [hunk 5]
```diff
@@ -26,9 +26,12 @@ export const postsApi = new postClient.PostsApi(
     basePath: 'https://pickify-posts-be-dev.m3ntorship.net/api',
   } as Configuration,
   undefined,
-  postsApiAxiosInstance,
 );
 
+export const postsApiTemp = axios.create({
```
Comment (Nilegfx) [inline modules/shared/api/postsApi.api.ts:31]: do we use this?

File: modules/shared/api/postsApi.api.ts [hunk 6]
```diff
@@ -26,9 +26,12 @@ export const postsApi = new postClient.PostsApi(
     basePath: 'https://pickify-posts-be-dev.m3ntorship.net/api',
   } as Configuration,
   undefined,
-  postsApiAxiosInstance,
```
Comment (Nilegfx) [inline modules/shared/api/postsApi.api.ts:29]: this change I am not sure about. if it works now without this instance we can merge it and come back to it later

**Review by Nilegfx — CHANGES_REQUESTED** [review]
good job! please address the PR comments.

File: modules/shared/api/postsApi.api.ts (hunk 5 above)
Comment (atwfk) [reply modules/shared/api/postsApi.api.ts:31]: no i will remove this as well

File: modules/shared/api/postsApi.api.ts (hunk 6 above)
Comment (atwfk) [reply modules/shared/api/postsApi.api.ts:29]: trust me :joy: 
it's useless since we are sending the Authorization with `getPosts` function

Comment (Nilegfx) [reply modules/shared/api/postsApi.api.ts:29]: not really. we might need it to add some other interceptor or defaults. but let's merge it like this for now.

Comment (atwfk) [reply modules/shared/api/postsApi.api.ts:29]: we will add another interceptor for this function but with no Authorization, i will test this solution later and let you know

File: modules/HomePage/components/Posts/PostsHelpers.ts (hunk 1 above)
Comment (atwfk) [reply modules/HomePage/components/Posts/PostsHelpers.ts]: done

File: modules/shared/api/getPosts.api.ts (hunk 2 above)
Comment (atwfk) [reply modules/shared/api/getPosts.api.ts]: done

File: modules/shared/api/getPosts.api.ts (hunk 3 above)
Comment (atwfk) [reply modules/shared/api/getPosts.api.ts]: done

File: modules/shared/api/getPosts.api.ts (hunk 4 above)
Comment (atwfk) [reply modules/shared/api/getPosts.api.ts]: done

**Review by Nilegfx — APPROVED** [review]
a7la LGTM 😍

**Pull request merged**